### ISSUE TITLE
Backblaze B2 Big File Copy stuck at 100%

### ISSUE THREAD

**Issue by intdel** [ncw/rclone#553]
I am using rclone 1.30 64bit on Linux Ubuntu 16.04.

I was migrating files from Amazon Cloud drive to Backblaze B2. It works fine with small files (e.g. 1GB), but when I was copying large files (e.g. 8GB ISO files) it would copy them, but never finish them. In the stats it will just show "100% done" and do nothing. It will even crash eventually.

The command I used was 
`./rclone copy acd:/ISO backblaze:FastnetBackup/ISO --transfers=8 --stats=3s`

then eventually it would just show

Linux/CentOS-7-x86_64-Everything-1503-01.iso: 100% done. avg: 6456.3, cur: 0.0 kByte/s.

However, instead of moving on to other transfers, it is stuck. This is a problem because all the other large ISOs get stuck as well.

**Comment by ncw** [comment]
Can you send me a log with `-v` of it happening please?

What do you mean by "It will even crash eventually" - was there an error message?

Can you copy one of those files from acd to local disk? And then from local disk to b2? Just trying to work out where the problem is!

Thanks

Nick

**Comment by intdel** [comment]
`root@ubuntu-512mb-sfo1-01:~/rclone-v1.30-linux-amd64# ./rclone copy acd:/ISO/Linux backblaze:FastnetBackup/ISO/Linux --transfers=8 --stats=10s -v
2016/06/30 12:26:16 rclone: Version "v1.30" starting with parameters ["./rclone" "copy" "acd:/ISO/Linux" "backblaze:FastnetBackup/ISO/Linux" "--transfers=8" "--stats=10s" "-v"]
2016/06/30 12:26:18 Linux: Failed to read metadata: object "Linux" not found
2016/06/30 12:26:18 Amazon cloud drive root 'ISO/Linux': Modify window not supported
2016/06/30 12:26:18 Amazon cloud drive root 'ISO/Linux': Building file list
2016/06/30 12:26:18 B2 bucket FastnetBackup path ISO/Linux/: Building file list
2016/06/30 12:26:18 Amazon cloud drive root 'ISO/Linux': Reading ""
2016/06/30 12:26:18 B2 bucket FastnetBackup path ISO/Linux/: Done building file list
2016/06/30 12:26:18 Amazon cloud drive root 'ISO/Linux': Finished reading ""
2016/06/30 12:26:18 Amazon cloud drive root 'ISO/Linux': Done building file list
2016/06/30 12:26:18 B2 bucket FastnetBackup path ISO/Linux/: Waiting for checks to finish
2016/06/30 12:26:18 Fedora-Live-Workstation-x86_64-23-10.iso: Sizes identical
2016/06/30 12:26:18 Fedora-Server-netinst-x86_64-23.iso: Sizes identical
2016/06/30 12:26:18 Fedora-Server-netinst-x86_64-23.iso: Unchanged skipping
2016/06/30 12:26:18 Fedora-Live-Workstation-x86_64-23-10.iso: Unchanged skipping
2016/06/30 12:26:18 ubuntu-16.04-server-amd64.iso: Sizes identical
2016/06/30 12:26:18 ubuntu-16.04-server-amd64.iso: Unchanged skipping
2016/06/30 12:26:18 CentOS-7-x86_64-LiveGNOME.iso: Sizes identical
2016/06/30 12:26:18 CentOS-7-x86_64-LiveGNOME.iso: Unchanged skipping
2016/06/30 12:26:18 ubuntu-16.04-desktop-amd64.iso: Sizes identical
2016/06/30 12:26:18 ubuntu-16.04-desktop-amd64.iso: Unchanged skipping
2016/06/30 12:26:18 F23-x86_64-CINN-20160512.iso: Sizes identical
2016/06/30 12:26:18 F23-x86_64-CINN-20160512.iso: Unchanged skipping
2016/06/30 12:26:18 B2 bucket FastnetBackup path ISO/Linux/: Waiting for transfers to finish
2016/06/30 12:26:19 CentOS-7-x86_64-Everything-1503-01.iso: Starting upload of large file in 76 chunks (id "4_z1a7a3e18055c9cd955580718_f201ce83af21ba2da_d20160630_m162619_c001_v0001029_t0022")
2016/06/30 12:26:24 CentOS-7-x86_64-Everything-1503-01.iso: Sending chunk 1 length 100663296
2016/06/30 12:26:28
Transferred:     141.140MBytes (11.904MByte/s)
Errors:                 0
Checks:                 6
Transferred:            0
Elapsed time:       11.8s
Transferring:
 *           CentOS-7-x86_64-Minimal-1503-01.iso:  7% done. avg: 5119.8, cur: 2719.2 kByte/s. ETA: 3m42s
 *        CentOS-7-x86_64-Everything-1503-01.iso:  1% done. avg: 11049.0, cur: 5166.6 kByte/s. ETA: 23m35s

2016/06/30 12:26:38
Transferred:     205.530MBytes (9.404MByte/s)
Errors:                 0
Checks:                 6
Transferred:            0
Elapsed time:       21.8s
Transferring:
 *           CentOS-7-x86_64-Minimal-1503-01.iso: 17% done. avg: 5894.4, cur: 4641.6 kByte/s. ETA: 1m56s
 *        CentOS-7-x86_64-Everything-1503-01.iso:  1% done. avg: 5202.1, cur: 2652.0 kByte/s. ETA: 45m58s

2016/06/30 12:26:48
Transferred:     282.187MBytes (8.858MByte/s)
Errors:                 0
Checks:                 6
Transferred:            0
Elapsed time:       31.8s
Transferring:
 *           CentOS-7-x86_64-Minimal-1503-01.iso: 29% done. avg: 6568.0, cur: 6223.2 kByte/s. ETA: 1m14s
 *        CentOS-7-x86_64-Everything-1503-01.iso:  1% done. avg: 3401.9, cur: 1361.2 kByte/s. ETA: 1h29m34s

2016/06/30 12:26:58
Transferred:     424.609MBytes (10.144MByte/s)
Errors:                 0
Checks:                 6
Transferred:            0
Elapsed time:       41.8s
Transferring:
 *           CentOS-7-x86_64-Minimal-1503-01.iso: 51% done. avg: 8621.9, cur: 10718.8 kByte/s. ETA: 29s
 *        CentOS-7-x86_64-Everything-1503-01.iso:  1% done. avg: 2527.3, cur:  698.7 kByte/s. ETA: 2h54m29s

2016/06/30 12:27:08
Transferred:     715.952MBytes (13.806MByte/s)
Errors:                 0
Checks:                 6
Transferred:            0
Elapsed time:       51.8s
Transferring:
 *           CentOS-7-x86_64-Minimal-1503-01.iso: 97% done. avg: 12948.3, cur: 20190.5 kByte/s. ETA: 0s
 *        CentOS-7-x86_64-Everything-1503-01.iso:  1% done. avg: 2010.4, cur:  358.6 kByte/s. ETA: 5h39m57s

2016/06/30 12:27:18
Transferred:         732MBytes (11.834MByte/s)
Errors:                 0
Checks:                 6
Transferred:            0
Elapsed time:      1m1.8s
Transferring:
 *           CentOS-7-x86_64-Minimal-1503-01.iso: 100% done. avg: 11033.1, cur: 10962.8 kByte/s. ETA: 0s
 *        CentOS-7-x86_64-Everything-1503-01.iso:  1% done. avg: 1669.1, cur:  184.1 kByte/s. ETA: 11h2m19s

2016/06/30 12:27:28
Transferred:     737.815MBytes (10.268MByte/s)
Errors:                 0
Checks:                 6
Transferred:            0
Elapsed time:     1m11.8s
Transferring:
 *           CentOS-7-x86_64-Minimal-1503-01.iso: 100% done. avg: 9434.8, cur: 5627.1 kByte/s. ETA: 0s
 *        CentOS-7-x86_64-Everything-1503-01.iso:  1% done. avg: 1513.3, cur:  369.2 kByte/s. ETA: 5h29m57s

2016/06/30 12:27:28 CentOS-7-x86_64-Everything-1503-01.iso: Received error: unexpected EOF - low level retry 1/10
2016/06/30 12:27:28 CentOS-7-x86_64-Everything-1503-01.iso: Removing failed copy
2016/06/30 12:27:29 CentOS-7-x86_64-Everything-1503-01.iso: Starting upload of large file in 76 chunks (id "4_z1a7a3e18055c9cd955580718_f201ce83af21ba2e1_d20160630_m162729_c001_v0001029_t0013")
2016/06/30 12:27:36 CentOS-7-x86_64-Everything-1503-01.iso: Sending chunk 1 length 100663296
2016/06/30 12:27:38
Transferred:     837.097MBytes (10.226MByte/s)
Errors:                 0
Checks:                 6
Transferred:            0
Elapsed time:     1m21.8s
Transferring:
 *           CentOS-7-x86_64-Minimal-1503-01.iso: 100% done. avg: 8240.9, cur: 2888.3 kByte/s. ETA: 0s
 *        CentOS-7-x86_64-Everything-1503-01.iso:  1% done. avg: 11940.6, cur: 5509.3 kByte/s. ETA: 22m7s
`

The error crash was some IO write error. I am not able to reproduce it. Might have been my mistake with choosing --transfers=40!

When I copy from local (ISO/Linux backblaze:FastnetBackup/Linux) it seems to work:
`2016/06/30 12:29:39 CentOS-7-x86_64-Everything-1503-01.iso: S**tarting upload of large file in 76 chunks** (id "4_z1a7a3e18055c9cd955580718_f20043bd69a6233f9_d20160630_m162939_c001_v0001029_t0044")
2016/06/30 12:29:41 CentOS-7-x86_64-Everything-1503-01.iso: Sending chunk 1 length 100663296
2016/06/30 12:29:47
Transferred:     140.312MBytes (1.552MByte/s)
Errors:                 0
Checks:                 6
Transferred:            0
Elapsed time:     1m30.4s
Transferring:
 *           CentOS-7-x86_64-Minimal-1503-01.iso:  6% done. avg:  545.5, cur:  397.4 kByte/s. ETA: 25m24s
 *        CentOS-7-x86_64-Everything-1503-01.iso:  1% done. avg: 12929.9, cur: 50553.4 kByte/s. ETA: 2m24s

2016/06/30 12:29:57
Transferred:     145.957MBytes (1.454MByte/s)
Errors:                 0
Checks:                 6
Transferred:            0
Elapsed time:     1m40.4s
Transferring:
 *           CentOS-7-x86_64-Minimal-1503-01.iso:  7% done. avg:  549.0, cur:  323.3 kByte/s. ETA: 30m56s
 *        CentOS-7-x86_64-Everything-1503-01.iso:  1% done. avg: 5584.6, cur: 25948.6 kByte/s. ETA: 4m41s
`

For some reason the upload to backblaze seems to be very slow at the moment. But I have verified earlier that copying from the local disk works for sure.

**Comment by ncw** [comment]
Thanks for that!

I can see something odd in the logs

```
2016/06/30 12:27:28 CentOS-7-x86_64-Everything-1503-01.iso: Received error: unexpected EOF - low level retry 1/10
2016/06/30 12:27:28 CentOS-7-x86_64-Everything-1503-01.iso: Removing failed copy
2016/06/30 12:27:29 CentOS-7-x86_64-Everything-1503-01.iso: Starting upload of large file in 76 chunks 
```

It looked like it retried the whole file, rather than retrying just the chunk which is odd.

I don't think that explains the getting stuck at 100% though - do you have a log of that?

**Comment by intdel** [comment]
Sadly no logs for the 100% issue.

**Comment by ncw** [comment]
Figured out what that log means - it means that reading from ACD gave the unexpected EOF error.

Any chance you could try to replicate the issue and get me a log with `-v`?

Thanks

**Comment by intdel** [comment]
It seems that the error is not showing up anymore. 
However, large file copies still seem to be very slow to B2. Not sure why that is. 

The 100% stuck error still appears though.

```
root@ubuntu-512mb-sfo1-01:~/rclone-v1.30-linux-amd64# ./rclone copy acd:/ISO/Linux backblaze:testbac/test/ -v --stats=10s
2016/07/01 11:39:41 rclone: Version "v1.30" starting with parameters ["./rclone" "copy" "acd:/ISO/Linux" "backblaze:testbac/test/" "-v" "--stats=10s"]
2016/07/01 11:39:43 test: Failed to read metadata: object "test" not found
2016/07/01 11:39:43 Amazon cloud drive root 'ISO/Linux': Modify window not supported
2016/07/01 11:39:43 Amazon cloud drive root 'ISO/Linux': Building file list
2016/07/01 11:39:43 Amazon cloud drive root 'ISO/Linux': Reading ""
2016/07/01 11:39:43 B2 bucket testbac path test/: Building file list
2016/07/01 11:39:43 B2 bucket testbac path test/: Done building file list
2016/07/01 11:39:44 Amazon cloud drive root 'ISO/Linux': Finished reading ""
2016/07/01 11:39:44 Amazon cloud drive root 'ISO/Linux': Done building file list
2016/07/01 11:39:44 B2 bucket testbac path test/: Waiting for checks to finish
2016/07/01 11:39:44 B2 bucket testbac path test/: Waiting for transfers to finish
2016/07/01 11:39:45 CentOS-7-x86_64-Everything-1503-01.iso: Starting upload of large file in 76 chunks (id "4_zaa7ace78156cac7955580718_f200b90fcd31add5b_d20160701_m153945_c001_v0001029_t0017")
2016/07/01 11:39:53
Transferred:     130.240MBytes (10.974MByte/s)
Errors:                 0
Checks:                 0
Transferred:            0
Elapsed time:       11.8s
Transferring:
 *                  F23-x86_64-CINN-20160512.iso:  7% done. avg: 10424.0, cur: 4836.7 kByte/s. ETA: 4m4s
 *                 ubuntu-16.04-server-amd64.iso:  0% done. avg:  622.4, cur: 1198.2 kByte/s. ETA: 9m15s
 *                ubuntu-16.04-desktop-amd64.iso:  1% done. avg: 2460.6, cur: 1645.9 kByte/s. ETA: 14m28s
 *        CentOS-7-x86_64-Everything-1503-01.iso:  0% done. avg: 1651.8, cur:  959.9 kByte/s. ETA: 2h8m28s

2016/07/01 11:40:03
Transferred:     435.211MBytes (19.902MByte/s)
Errors:                 0
Checks:                 0
Transferred:            0
Elapsed time:       21.8s
Transferring:
 *                  F23-x86_64-CINN-20160512.iso: 21% done. avg: 14587.7, cur: 11248.3 kByte/s. ETA: 1m28s
 *                 ubuntu-16.04-server-amd64.iso:  1% done. avg:  431.1, cur:  730.5 kByte/s. ETA: 15m7s
 *                ubuntu-16.04-desktop-amd64.iso:  6% done. avg: 4986.8, cur: 4417.5 kByte/s. ETA: 5m6s
 *        CentOS-7-x86_64-Everything-1503-01.iso:  0% done. avg: 3689.7, cur: 3089.0 kByte/s. ETA: 39m37s

2016/07/01 11:40:06 CentOS-7-x86_64-Everything-1503-01.iso: Sending chunk 1 length 100663296
2016/07/01 11:40:13
Transferred:     738.761MBytes (23.181MByte/s)
Errors:                 0
Checks:                 0
Transferred:            0
Elapsed time:       31.8s
Transferring:
 *                  F23-x86_64-CINN-20160512.iso: 34% done. avg: 15018.3, cur: 13649.4 kByte/s. ETA: 1m1s
 *                 ubuntu-16.04-server-amd64.iso:  8% done. avg: 1960.6, cur: 2756.1 kByte/s. ETA: 3m43s
 *                ubuntu-16.04-desktop-amd64.iso: 11% done. avg: 5847.6, cur: 5949.2 kByte/s. ETA: 3m35s
 *        CentOS-7-x86_64-Everything-1503-01.iso:  1% done. avg: 3445.9, cur: 2947.0 kByte/s. ETA: 41m22s

2016/07/01 11:40:23
Transferred:       1.086GBytes (26.568MByte/s)
Errors:                 0
Checks:                 0
Transferred:            0
Elapsed time:       41.8s
Transferring:
 *                  F23-x86_64-CINN-20160512.iso: 48% done. avg: 15913.0, cur: 16097.9 kByte/s. ETA: 40s
 *                 ubuntu-16.04-server-amd64.iso: 22% done. avg: 3982.9, cur: 6341.4 kByte/s. ETA: 1m21s
 *                ubuntu-16.04-desktop-amd64.iso: 18% done. avg: 6935.1, cur: 8125.5 kByte/s. ETA: 2m25s
 *        CentOS-7-x86_64-Everything-1503-01.iso:  1% done. avg: 2552.0, cur: 1512.7 kByte/s. ETA: 1h20m36s

2016/07/01 11:40:33
Transferred:       1.538GBytes (30.357MByte/s)
Errors:                 0
Checks:                 0
Transferred:            0
Elapsed time:       51.8s
Transferring:
 *                  F23-x86_64-CINN-20160512.iso: 64% done. avg: 16755.2, cur: 18214.3 kByte/s. ETA: 24s
 *                 ubuntu-16.04-server-amd64.iso: 35% done. avg: 4882.8, cur: 7393.5 kByte/s. ETA: 58s
 *                ubuntu-16.04-desktop-amd64.iso: 30% done. avg: 9081.8, cur: 12738.2 kByte/s. ETA: 1m18s
 *        CentOS-7-x86_64-Everything-1503-01.iso:  1% done. avg: 2332.1, cur: 1476.2 kByte/s. ETA: 1h22m25s

2016/07/01 11:40:38 CentOS-7-x86_64-Everything-1503-01.iso: Sending chunk 2 length 100663296
2016/07/01 11:40:43
Transferred:       1.970GBytes (32.602MByte/s)
Errors:                 0
Checks:                 0
Transferred:            0
Elapsed time:      1m1.8s
Transferring:
 *                  F23-x86_64-CINN-20160512.iso: 73% done. avg: 15924.2, cur: 15108.9 kByte/s. ETA: 22s
 *                 ubuntu-16.04-server-amd64.iso: 50% done. avg: 5803.5, cur: 8805.1 kByte/s. ETA: 37s
 *                ubuntu-16.04-desktop-amd64.iso: 40% done. avg: 10064.0, cur: 13639.6 kByte/s. ETA: 1m2s
 *        CentOS-7-x86_64-Everything-1503-01.iso:  2% done. avg: 3359.5, cur: 4469.1 kByte/s. ETA: 26m54s

2016/07/01 11:40:53
Transferred:       2.432GBytes (34.651MByte/s)
Errors:                 0
Checks:                 0
Transferred:            0
Elapsed time:     1m11.8s
Transferring:
 *                  F23-x86_64-CINN-20160512.iso: 85% done. avg: 15793.8, cur: 15176.8 kByte/s. ETA: 12s
 *                 ubuntu-16.04-server-amd64.iso: 75% done. avg: 7458.5, cur: 12925.6 kByte/s. ETA: 12s
 *                ubuntu-16.04-desktop-amd64.iso: 52% done. avg: 10973.5, cur: 15021.5 kByte/s. ETA: 46s
 *        CentOS-7-x86_64-Everything-1503-01.iso:  2% done. avg: 2869.3, cur: 2293.9 kByte/s. ETA: 52m26s

2016/07/01 11:41:03
Transferred:       2.899GBytes (36.252MByte/s)
Errors:                 0
Checks:                 0
Transferred:            0
Elapsed time:     1m21.8s
Transferring:
 *                  F23-x86_64-CINN-20160512.iso: 99% done. avg: 16052.6, cur: 16579.1 kByte/s. ETA: 0s
 *                 ubuntu-16.04-server-amd64.iso: 99% done. avg: 8513.9, cur: 14255.8 kByte/s. ETA: 0s
 *                ubuntu-16.04-desktop-amd64.iso: 62% done. avg: 11525.2, cur: 15057.3 kByte/s. ETA: 35s
 *        CentOS-7-x86_64-Everything-1503-01.iso:  2% done. avg: 2503.6, cur: 1177.5 kByte/s. ETA: 1h42m9s

2016/07/01 11:41:12 CentOS-7-x86_64-Everything-1503-01.iso: Sending chunk 3 length 100663296
2016/07/01 11:41:13
Transferred:       3.100GBytes (34.553MByte/s)
Errors:                 0
Checks:                 0
Transferred:            0
Elapsed time:     1m31.8s
Transferring:
 *                  F23-x86_64-CINN-20160512.iso: 100% done. avg: 14329.9, cur: 9188.8 kByte/s. ETA: 0s
 *                 ubuntu-16.04-server-amd64.iso: 100% done. avg: 7601.8, cur: 7605.6 kByte/s. ETA: 0s
 *                ubuntu-16.04-desktop-amd64.iso: 69% done. avg: 11366.5, cur: 12566.8 kByte/s. ETA: 34s
 *        CentOS-7-x86_64-Everything-1503-01.iso:  3% done. avg: 3331.4, cur: 5843.8 kByte/s. ETA: 20m18s

2016/07/01 11:41:23
Transferred:       3.237GBytes (32.536MByte/s)
Errors:                 0
Checks:                 0
Transferred:            0
Elapsed time:     1m41.8s
Transferring:
 *                  F23-x86_64-CINN-20160512.iso: 100% done. avg: 12880.7, cur: 4716.5 kByte/s. ETA: 0s
 *                 ubuntu-16.04-server-amd64.iso: 100% done. avg: 6828.4, cur: 3903.9 kByte/s. ETA: 0s
 *                ubuntu-16.04-desktop-amd64.iso: 79% done. avg: 11665.8, cur: 13577.3 kByte/s. ETA: 21s
 *        CentOS-7-x86_64-Everything-1503-01.iso:  3% done. avg: 2993.5, cur: 2999.5 kByte/s. ETA: 39m33s
```


<snip>


```
2016/07/01 11:42:55 CentOS-7-x86_64-Everything-1503-01.iso: Sending chunk 6 length 100663296
2016/07/01 11:43:03
Transferred:       3.800GBytes (19.274MByte/s)
Errors:                 0
Checks:                 0
Transferred:            0
Elapsed time:     3m21.8s
Transferring:
 *                  F23-x86_64-CINN-20160512.iso: 100% done. avg: 6401.8, cur:    6.0 kByte/s. ETA: 0s
 *                 ubuntu-16.04-server-amd64.iso: 100% done. avg: 3383.5, cur:    5.0 kByte/s. ETA: 0s
 *                ubuntu-16.04-desktop-amd64.iso: 100% done. avg: 7289.7, cur:   63.9 kByte/s. ETA: 0s
 *        CentOS-7-x86_64-Everything-1503-01.iso:  7% done. avg: 2971.0, cur: 3139.6 kByte/s. ETA: 36m13s



```

**Comment by intdel** [comment]
Looks like if I wait long enough, it eventually finishes the 100% thing:

```
2016/07/01 11:44:13
Transferred:       3.987GBytes (15.018MByte/s)
Errors:                 0
Checks:                 0
Transferred:            0
Elapsed time:     4m31.8s
Transferring:
 *                  F23-x86_64-CINN-20160512.iso: 100% done. avg: 4734.8, cur:    0.1 kByte/s. ETA: 0s
 *                 ubuntu-16.04-server-amd64.iso: 100% done. avg: 2500.5, cur:    0.0 kByte/s. ETA: 0s
 *                ubuntu-16.04-desktop-amd64.iso: 100% done. avg: 5393.1, cur:    0.6 kByte/s. ETA: 0s
 *        CentOS-7-x86_64-Everything-1503-01.iso: 10% done. avg: 2928.7, cur: 2077.2 kByte/s. ETA: 53m10s

2016/07/01 11:44:18 ubuntu-16.04-server-amd64.iso: Copied (new)
2016/07/01 11:44:23
Transferred:       4.113GBytes (14.941MByte/s)
Errors:                 0
Checks:                 0
Transferred:            1
Elapsed time:     4m41.8s
Transferring:
 *                  F23-x86_64-CINN-20160512.iso: 100% done. avg: 4565.0, cur:    0.0 kByte/s. ETA: 0s
 *                 CentOS-7-x86_64-LiveGNOME.iso:  4% done. avg: 12823.2, cur: 3940.1 kByte/s. ETA: 4m48s
 *                ubuntu-16.04-desktop-amd64.iso: 100% done. avg: 5200.0, cur:    0.3 kByte/s. ETA: 0s
 *        CentOS-7-x86_64-Everything-1503-01.iso: 11% done. avg: 3107.6, cur: 4957.6 kByte/s. ETA: 22m0s

2016/07/01 11:44:24 CentOS-7-x86_64-Everything-1503-01.iso: Sending chunk 9 length 100663296
2016/07/01 11:44:24 pacer: Rate limited, increasing sleep to 20ms
2016/07/01 11:44:25 pacer: low level retry 1/10 (error c001_v0001029_t0043 is too busy (503 service_unavailable))
2016/07/01 11:44:25 CentOS-7-x86_64-Everything-1503-01.iso: Sending chunk 9 length 100663296
2016/07/01 11:44:25 pacer: Rate limited, increasing sleep to 40ms
2016/07/01 11:44:25 pacer: low level retry 2/10 (error c001_v0001029_t0043 is too busy (503 service_unavailable))
2016/07/01 11:44:25 CentOS-7-x86_64-Everything-1503-01.iso: Sending chunk 9 length 100663296
2016/07/01 11:44:25 pacer: Rate limited, increasing sleep to 80ms
2016/07/01 11:44:25 pacer: low level retry 3/10 (error c001_v0001029_t0043 is too busy (503 service_unavailable))
2016/07/01 11:44:25 CentOS-7-x86_64-Everything-1503-01.iso: Sending chunk 9 length 100663296
2016/07/01 11:44:25 pacer: Rate limited, increasing sleep to 160ms
2016/07/01 11:44:25 pacer: low level retry 4/10 (error c001_v0001029_t0043 is too busy (503 service_unavailable))
2016/07/01 11:44:25 CentOS-7-x86_64-Everything-1503-01.iso: Sending chunk 9 length 100663296
2016/07/01 11:44:30 pacer: Reducing sleep to 120ms
2016/07/01 11:44:30 ubuntu-16.04-desktop-amd64.iso: Copied (new)
2016/07/01 11:44:33
Transferred:       4.261GBytes (14.951MByte/s)
Errors:                 0
Checks:                 0
Transferred:            2
Elapsed time:     4m51.8s
Transferring:
 *                  F23-x86_64-CINN-20160512.iso: 100% done. avg: 4406.9, cur:    0.0 kByte/s. ETA: 0s
 *                 CentOS-7-x86_64-LiveGNOME.iso: 14% done. avg: 12383.3, cur: 7767.7 kByte/s. ETA: 2m10s
 *        CentOS-7-x86_64-Everything-1503-01.iso: 11% done. avg: 3066.4, cur: 3726.4 kByte/s. ETA: 29m12s
 *      Fedora-Live-Workstation-x86_64-23-10.iso:  1% done. avg: 4773.4, cur: 1649.7 kByte/s. ETA: 14m20s

```

**Comment by ncw** [comment]
I've realised this is the same issue as #/587 - it is to do with the internal buffering on large file uploads for B2.  We read the file into memory in 96MB chunks very quickly - then while we upload the last of these chunks the transfer speed drops to 0.

You can see from the `Transferred:       3.100GBytes (34.553MByte/s)` lines that everything is proceeding properly.

@klauspost we discussed this before didn't we - did we ever make an issue about it?

**Comment by intdel** [comment]
That makes sense.

So would changing the chunk size make this issue more tolerable?

**Comment by ncw** [comment]
It is purely a display problem - the transfer is happening properly. Unfortunately rclone already uses the minimum chunk size so you can't make it smaller (which would help). 

One thing you could try is to set a --bwlimit of about the capacity of your Internet connection - that will probably work. 

I need to think of a better way of accounting the transfers so they display properly in this case.

**Comment by ncw** [comment]
The display problem was fixed as part of #602 and is now in the v1.33 release!
